### ISSUE TITLE
Allow ops to take several zero copy buffers as arguments

### ISSUE THREAD

**Issue by SyrupThinker** [denoland/deno#4786]
# Problem
Currently a op can only optionally take one zero copy buffer.
Because of this multiple buffers need to be merged into one to pass them to Rust,
forcing several copies. (This could also be done by preserving data between calls, which can be considered an even worse solution)
```ts
function sign_a(secret: Uint8Array, data: Uint8Array) {
  const single_buffer = /* Copy secret and data into single_buffer */;
  return sign_op.dispatch(new Uint8Array([...]), single_buffer);
}
```
```rust
fn sign_op(control: &[u8], zero_copy: Option<ZeroCopyBuf>) -> CoreOp {
  // Split zero_copy into secret and data
  return Op::Sync(somemodule::sign(secret, data) /* => Box<[u8]> */);
}
```

# Proposal
Allow ops to take several zero copy buffers as arguments.
This eliminates the copies and makes the plugin API easier to use for multi-buffer inputs.

```ts
function sign_a(secret: Uint8Array, data: Uint8Array) {
  return sign_op.dispatch(new Uint8Array([...]), secret, data);
}
```
```rust
fn sign_op(control: &[u8], zero_copy: Box<[ZeroCopyBuf]>) -> CoreOp {
  if (zero_copy.len() != 2) {
    return; // Signal error
  }
  // Handle control ...
  return Op::Sync(somemodule::sign(zero_copy[0], zero_copy[1]) /* => Box<[u8]> */);
}
```

**Comment by ry** [comment]
I'm in favor of this feature. It has been something I always thought we would do if we needed it.

The only caveat is that it must be done in a way that does not degrade performance. Since this is touching some very hot interfaces, it will be important to carefully validate that it does not introduce regressions.

**Comment by SyrupThinker** [comment]
As noted in PR #4788 it needs to be decided how the buffers are exposed on the Rust side.
The options I considered are `Box<[ZeroCopyBuf]>` (what the PR currently implements), `&[ZeroCopyBuf]` and `&mut [Option<ZeroCopyBuf>]` and some rationale can be found [here](https://github.com/denoland/deno/pull/4788#issue-404762399).

**Comment by SyrupThinker** [comment]
Resolved with becbb56b19e96e4dd72b861217a855fba953d290
